supervisor/store/repository.py

```diff
@@ -103,9 +105,17 @@ async def load(self) -> None:
 
     async def update(self) -> None:
         """Update add-on repository."""
-        if self.type == StoreType.LOCAL or not self.validate():
+        if not self.validate():
             return
-        await self.git.pull()
+
+        if self.type == StoreType.LOCAL or await self.git.pull():
+            await asyncio.gather(
+                *[
+                    addon.refresh_cache()
+                    for addon in self.sys_addons.all
+                    if addon.repository == self.slug
```

Comment:
The `Repository` class so far did not really deal with individual add-ons, maybe it would be better if we update cache from `StoreManager` instead? :thinking:  Then we have the `refresh_cache` calls at the same place. :thinking:

Reply:
Well we don't always call it from `StoreManager` currently. We also call `Repository.update` from here:
https://github.com/home-assistant/supervisor/blob/749d0e6c27d2a4ed88e9018ad657b26a4a6bb1d2/supervisor/resolution/fixups/store_execute_reload.py#L44

Also the `update` method is pretty standard on our classes and always returns `None`. This would have to return something else to let the caller know they need to do more. I feel like the repository is the thing that updated so it should just handle it.

Reply:
> Well we don't always call it from StoreManager currently. We also call Repository.update from here:

Right, which makes me wonder if that works correctly there:  It doesn't update `StoreManager`, so what if the repository has new/deleted/updated add-ons compared to before the fixups have become necessary? :thinking: 

Anyways, this would be a different PR.

> Also the `update` method is pretty standard on our classes and always returns `None`.

Sure, standardization is nice if wee have good reasons. But we also deviate from it (e.g. in `MountManager`/`Mount` implementation). And I think we should deviate, if we have good reasons.

Actually, looking a bit more into this code, I think we should not over-optimize these four stat calls here.

`StoreManager`'s `reload()` calls `self.data.update()` through `self.load()`, which causes a complete read of all the add-ons in `_read_addons_folder()`.

Couldn't we just add `addon_config[ATTR_WITH_LOGO]` etc. to  `_read_addons_folder()`? :thinking: 

I really like the idea of avoiding updating the whole store when git pull doesn't change something. But I think we should apply this to the complete store update *somehow*. But this would be material for a different PR.

Reply:
We can tweak this in a small way if we route everything through store manager. Since we'll know when a git repo has updated, we can skip all the other stuff if there were no updates. In future we could also be more selective about only re-reading in the specific git repos that updated.

Reply:
> Well no, it has to be the same PR actually. This PR relies on the fact that whenever we actually detect a change in commit in the repo, we refresh our cache. 

My suggestion is, in a first step, just always update the cache (or rather store them as `addon_config` in `_read_addons_folder()` just like all the other add-on related stuff.

In a second step, we then can opt out of refresh these data as well. Which will lead to even less I/O.

But I am also ok doing it in one step. :smile:

Reply:
@agners take a look, I think I incorporated this in a reasonable way for the time being. The logic is all in `StoreManager` as discussed. Also as it turns out, we were swallowing any errors that occurred during `Repository.update` so now we're at least logging those. Hazards of `asyncio.wait` vs `asyncio.gather` and their differences 😬

Reply:
Awesome thanks! LGTM!

> Hazards of asyncio.wait vs asyncio.gather and their differences 😬

Ugh I see, nice catch!